src/yarp.c

```diff
@@ -10030,6 +10064,44 @@ parse_variable_call(yp_parser_t *parser) {
             return (yp_node_t *) yp_local_variable_read_node_create(parser, &parser->previous, (uint32_t) depth);
         }
 
+        if (!parser->current_scope->closed && token_is_numbered_parameter(parser->previous.start, parser->previous.end)) {
+            // Now that we know we have a numbered parameter, we need to check
+            // if it's allowed in this context. If it is, then we will create a
+            // local variable read. If it's not, then we'll create a normal call
+            // node but add an error.
+            if (parser->current_scope->explicit_params) {
```

Comment:
Should we store information about the current scope not being able to have children with numbered params in all of these cases and not only the happy path? We already know that the current scope can't have children with scoped params. But any children will still have to look past this scope to see that, even though we know the information now. 

I know this will only happen in the error case, and am unsure how much we're trying to optimize that / error recovery.

Reply:
Sorry I'm not sure I'm following. There's two errors this thing is catching. One is:

```ruby
foo { || _1 }
```

is disallowed because of the combination of explicit and implicit parameters. The other is:

```ruby
foo { _1; bar { _1 } }
```

which is that you can't use implicit parameters in a block that is inside another scope that is using implicit parameters. We need both sets of information to stop both of these.

I'm not sure if that answered your question.

Reply:
@jemmaissroff I see what you're saying now. We cache this information on the current scope and check it here: https://github.com/ruby/yarp/pull/1466/files#diff-71ae829a627932e547cf825850d9a6b4773b009b0b68ae8eb1d8ef02e6d98987R10126. So that function bails early.

Reply:
Wait no, that isn't what you were saying. Fixing now.